core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -184,7 +184,9 @@ void *impl_allocate_common(const int device_id,
   cudaError_t error_code = cudaSuccess;
 #ifndef CUDART_VERSION
 #error CUDART_VERSION undefined!
-#elif (defined(KOKKOS_ENABLE_IMPL_CUDA_MALLOC_ASYNC) && CUDART_VERSION >= 11020)
+#elif (defined(KOKKOS_ENABLE_IMPL_CUDA_MALLOC_ASYNC) && \
+       CUDART_VERSION >= 11020 &&                       \
+       !defined(KOKKOS_ENABLE_IMPL_CUDA_UNIFIED_MEMORY))
```

Comment:
Why not just
```suggestion
#elif defined(KOKKOS_ENABLE_IMPL_CUDA_EMULATE_UNIFIED_MEMORY)
  error_code = cudaMallocManaged(&ptr, arg_alloc_size, cudaMemAttachGlobal);
#else
#if (defined(KOKKOS_ENABLE_IMPL_CUDA_MALLOC_ASYNC) && \
       CUDART_VERSION >= 11020 &&                       \
       !defined(KOKKOS_ENABLE_IMPL_CUDA_UNIFIED_MEMORY))
```
and adding another `#endif` below? We only need the extra scope for deciding between `cudaMalloc` and `cudaMallocAsync` but in the new case we always want to use `cudaMallocManaged`.